### PR TITLE
Use panel for Rewards page onboarding

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/24538

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Open the browser with a new profile.
- Navigate to `brave://rewards`.
- Click the `Start Using Rewards` button.
- Verify that the Rewards panel is opened with onboarding displayed.
- Click "Start Using Rewards" on the panel.
- Close the panel.
- Verify that Rewards page content is displayed.

### CODE REVIEW

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 1]
```diff
@@ -2046,14 +2051,14 @@ void RewardsDOMHandler::GetOnboardingStatus(const base::Value::List& args) {
                          base::Value(std::move(data)));
 }
 
-void RewardsDOMHandler::SaveOnboardingResult(const base::Value::List& args) {
-  CHECK_EQ(1U, args.size());
-  if (!rewards_service_)
-    return;
-
+void RewardsDOMHandler::EnableRewards(const base::Value::List& args) {
```
Comment (emerick) [inline browser/ui/webui/brave_rewards_page_ui.cc:2054]: It doesn't seem like this function does anything on Android. Are we going to open the Android-native panel via a different mechanism? Or is this desktop-only for now?

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 2]
```diff
@@ -59,6 +59,21 @@ using content::WebUIMessageHandler;
...
+
+brave_rewards::RewardsPanelCoordinator* GetPanelCoordinator(
+    content::WebContents* web_contents) {
+  DCHECK(web_contents);
+  auto* browser = chrome::FindBrowserWithWebContents(web_contents);
+  if (!browser) {
+    return nullptr;
+  }
+
+  return brave_rewards::RewardsPanelCoordinator::FromBrowser(browser);
```
Comment (szilardszaloki) [inline browser/ui/webui/brave_rewards_page_ui.cc:72]: Nit:
```
if (auto* browser = chrome::FindBrowserWithWebContents(web_contents)) {
  return brave_rewards::RewardsPanelCoordinator::FromBrowser(browser);
}
```

File: browser/ui/webui/brave_rewards_page_ui.cc (hunk 1 above)
Comment (zenparsing) [reply browser/ui/webui/brave_rewards_page_ui.cc:2054]: On Android, there's another mechanism to that shows a native onboarding modal when the Rewards page is accessed. I'll add some comments here to explain.

**Review by szilardszaloki — APPROVED** [review]

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 3]
```diff
@@ -2036,24 +2039,27 @@ void RewardsDOMHandler::OnGetWalletPassphrase(const std::string& passphrase) {
...
                          base::Value(std::move(data)));
 }
 
-void RewardsDOMHandler::SaveOnboardingResult(const base::Value::List& args) {
-  CHECK_EQ(1U, args.size());
-  if (!rewards_service_)
-    return;
-
+void RewardsDOMHandler::EnableRewards(const base::Value::List& args) {
   AllowJavascript();
```
Comment (emerick) [inline browser/ui/webui/brave_rewards_page_ui.cc:2052]: Totally minor, but should we just move this into the `!BUILDFLAG(IS_ANDROID)` section?

**Review by emerick — APPROVED** [review]
LGTM, just mentioned one super-tiny nit.

File: browser/ui/webui/brave_rewards_page_ui.cc (hunk 3 above)
Comment (zenparsing) [reply browser/ui/webui/brave_rewards_page_ui.cc]: Done.

**Pull request merged**